src/devices/src/virtio/block/device.rs

```diff
@@ -384,6 +379,23 @@ impl Block {
         }
     }
 
+    pub fn process_async_completion_event(&mut self) {
+        let engine = match self.disk.file_engine_mut() {
+            FileEngine::Async(engine) => engine,
+            FileEngine::Sync(_) => {
+                error!("The block device doesn't use an async IO engine");
+                return;
+            }
+        };
```

Comment:
The same logic is also defined in `process_async_completion_queue`. Should we have a dedicated function for this part of code, instead of duplicating it?

Reply:
Makes sense. Done.